src/frontier_util.c

```diff
@@ -2474,7 +2485,12 @@ void CopyFrontierBrainTrainerName(u8 *dst)
     else
         facility = VarGet(VAR_FRONTIER_FACILITY);
 
+#ifndef BUGFIX
+    // Incorrectly reads up to one character less than the player name length instead of the full trainer name length.
     for (i = 0; i < PLAYER_NAME_LENGTH; i++)
+#else
+    for (i = 0; i < TRAINER_NAME_LENGTH + 1; i++)
+#endif
         dst[i] = gTrainers[sFrontierBrainTrainerIds[facility]].trainerName[i];
 
     dst[i] = EOS;
```

Comment:
Oh, then it might mean that it's a new bug introduced in non-English versions, as this is what changed in French/Italian versions:

```diff
    s32 i;
    s32 facility;

    if (gBattleTypeFlags & BATTLE_TYPE_RECORDED)
        facility = GetRecordedBattleFrontierFacility();
    else
        facility = VarGet(VAR_FRONTIER_FACILITY);

-    for (i = 0; i < PLAYER_NAME_LENGTH; i++)
+    for (i = 0; i < TRAINER_NAME_LENGTH + 1; i++)
        dst[i] = gTrainers[sFrontierBrainTrainerIds[facility]].trainerName[i];

    dst[i] = EOS;
```